src/formatDate.js

```diff
@@ -8,7 +8,29 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const newDate = [];
+  const dateSet = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateSet[fromFormat[i]] = dateParts[i];
+  }
+
+  if (dateSet.YYYY) {
+    dateSet.YY = dateSet.YYYY.slice(2);
+  }
+
+  if (dateSet.YY < 30) {
+    dateSet.YYYY = '20' + `${dateSet.YY}`;
+  } else {
+    dateSet.YYYY = '19' + `${dateSet.YY}`;
+  }
```

Comment:
The logic for converting the year from 'YYYY' to 'YY' and vice versa is not correctly placed. It should be executed only if the respective format is present in the 'toFormat' array. Also, the conversion from 'YY' to 'YYYY' should only happen if 'YY' is in the 'fromFormat' and 'YYYY' is in the 'toFormat'. Currently, it is executed unconditionally, which could lead to incorrect results if 'YY' or 'YYYY' is not part of the 'toFormat'.